connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -338,6 +340,22 @@ class SparkConnectPlanner(session: SparkSession) {
     }
   }
 
+  private def transformCollectMetrics(rel: proto.CollectMetrics): LogicalPlan = {
+    val metrics = rel.getMetricsList.asScala.map { expr =>
+      Column(transformExpression(expr))
+    }
+
+    if (rel.getIsObservation) {
```

Comment:
What is the difference between the code paths?

Reply:
The Observation registers `ObservationListener` on `ExecutionListenerManager`.

Reply:
This explanation is not fully correct. The `Observation` class uses event listeners to be able to fetch the metrics as soon as they appear without waiting for the DS command to finish. Since we're not having an event listener at this point in time it's not the same thing. Please simplify the PR accordingly.

Reply:
We don't need Observation here. We just need to send the observed metrics as part of the response stream.

Reply:
OK. I got it.

Reply:
OK. Let only support the string.

Reply:
> We don't need Observation here. We just need to send the observed metrics as part of the response stream.

But we should maintain the consistency of behavior between the API of spark connect and the API of pyspark. The observe of pyspark supports using `Observation` as parameter and the doc test checks the consistence.

Maybe we could keep the API of connect supports `Observation` and it will not be used at server side, but directly using `CollectMetrics`.

Reply:
Please remove the `is_observation` code path.

Reply:
@hvanhovell is_observation has been removed.